chartfx-math/src/main/java/de/gsi/math/ArrayMath.java

```diff
@@ -180,15 +275,17 @@
      * filter from: "The Scientist and Engineer's Guide to DSP" Chapter 20
      *
      * @param signal input signal
-     * @param cutoffFraction cutoffFreq must be smaller than half the samplerate
+     * @param output output signal (if null a new vector is being allocated)
+     * @param cutoffFraction cutoffFreq must be smaller than half the sample rate
      * @param filterOrder can be between 2 and 20
      * @param filterType 0: lowPass 1: highPass
      * @param ripplePercent ripplePercent is amount of ripple in Chebyshev filter (0-29) (ripplePercent == 0 -&gt;
-     *        Butterworth)
+     *            Butterworth)
      * @return filtered output signal
      */
-    public static double[] filterSignal(final double[] signal, final double cutoffFraction, final double filterOrder,
-            final FilterType filterType, final double ripplePercent) {
+    public static double[] filterSignal(final double[] signal, final double[] output, final double cutoffFraction, final int filterOrder, final FilterType filterType, final double ripplePercent) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'filterSignal(double, double, double, int, FilterType, double)' has an NPath complexity of 3200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5080443)